tests/v3_api/test_registry.py

```diff
@@ -0,0 +1,189 @@
+from common import *   # NOQA
+import pytest
+
+namespace = {"p_client": None, "ns": None, "cluster": None, "project": None}
+REGISTRY_USER_NAME = os.environ.get('REGISTRY_USER_NAME', "None")
+REGISTRY_PASSWORD = os.environ.get('REGISTRY_PASSWORD', "None")
+TEST_CLIENT_IMAGE = os.environ.get('TEST_CLIENT_IMAGE', "None")
+REGISTRY = os.environ.get('REGISTRY', "None")
+
+
+def test_create_registry_single_namespace():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+
+    create_registry_validate_workload(p_client, ns)
+
+
+def test_create_registry_all_namespace():
+    p_client = namespace["p_client"]
+    c_client = namespace["c_client"]
+    cluster = namespace["cluster"]
+    project = namespace["project"]
+
+    ns = namespace["ns"]
+    create_registry_validate_workload(p_client, ns, allns=True)
+
+    # Create and validate workload in a new namespace
+    new_ns = create_ns(c_client, cluster, project)
+    create_validate_workload(p_client, new_ns)
+
+
+def test_delete_registry_all_namespace():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+
+    registry = create_registry_validate_workload(p_client, ns, allns=True)
+    delete_registry(p_client, registry)
+
+
+def test_delete_registry_single_namespace():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+
+    registry = create_registry_validate_workload(p_client, ns)
+    delete_registry(p_client, registry)
+
```

Comment:
After delete_registry, make sure that workload deployment with private images should fails in the namespace where it was present earlier.